### PR TITLE
feat: Clarify numerical validation errors

### PR DESCRIPTION
**Pull request by ypnos**
Current messages for `minimum` and `maximum` were misleading for a wide audience. Messages are now changed to "at least" for `minimum` and "at most" for `maximum`.
Code simplified.

### CODE REVIEW

File: lib/jsonschema.lua [hunk 1]
```diff
@@ -895,34 +895,23 @@ generate_validator = function(ctx, schema)
   if schema.minimum or schema.maximum or schema.multipleOf or schema.exclusiveMinimum or schema.exclusiveMaximum then
     ctx:stmt(sformat('if %s == "number" then', datatype))
 
-    if schema.minimum then
-      local op = '<'
-      local msg = 'greater'
-      ctx:stmt(sformat('  if %s %s %s then', ctx:param(1), op, schema.minimum))
-      ctx:stmt(sformat('    return false, %s("expected %%s to be %s than %s", %s)',
-                       ctx:libfunc('string.format'), msg, schema.minimum, ctx:param(1)))
+    local addRangeCheck = function (op, reference, msg)
```
Comment (membphis) [inline lib/jsonschema.lua:898]: This way of writing has a bad performance. You can use this way:

```lua
local function(ctx, op, reference, msg)
   ... ...
end

generate_validator = function(ctx, schema)
    ...
    if schema.minimum then
      addRangeCheck(ctx, '<', schema.minimum, 'at least')
    end
    ...
end

**Review by membphis — APPROVED** [review]

**Review by membphis — CHANGES_REQUESTED** [review]
please change to new way

**Comment by membphis** [comment]
ping @ypnos

File: lib/jsonschema.lua (hunk 1 above)
Comment (ypnos) [reply lib/jsonschema.lua]: Sorry for delay. I incorporated your improvement.

**Review by membphis — APPROVED** [review]

**Comment by membphis** [comment]
@spacewander do you have time to look at this PR? many thx

**Review by spacewander — APPROVED** [review]

**Pull request merged**